### PR TITLE
Android: Add file browser screen to new UI, and several tweaks.

### PR DESCRIPTION
**Pull request by sigmabeta**
I added the file browser screen with effectively the same functionality as the old version - that is, you click on folders until you find an ISO, then the app knows that folder is the one to check for games on startup. This will eventually be replaced with functionality that will index all known games into an SQLite database so we don't have to keep calling native methods all the time; but I implemented this much because without it, the new UI doesn't actually do very much.

I also implemented a bunch of UI tweaks; for example, all touchable things now have ripple feedback; the  game cards no longer have a details button (the properties dialog is now accessible via long-click), Picasso now resizes screenshots in memory based on their displayed size (this reduces memory usage with 5 games on screen from ~55MB to ~10MB,) and more.

Since I think everyone kind of gets the idea of what's going on, I'm attaching any further screenshots at ~1/4 size. If you want the full versions, let me know.

grid-games

Note the floating action button, which triggers the file browser.

add-folders

This screen will also have a FAB, so the user can click that instead of the target file (since its kind of weird to click on a single file when you want to add a whole folder.)

add-games

Anything touchable will have a cool looking ripple emanate from the touchpoint, which indicates both actionability and long-clickability (if present).

### CODE REVIEW

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/activities/AddDirectoryActivity.java [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+		// Specifying the LayoutManager determines how the RecyclerView arranges views.
+		RecyclerView.LayoutManager layoutManager = new LinearLayoutManager(this, LinearLayoutManager.VERTICAL, false);
+		recyclerView.setLayoutManager(layoutManager);
+
+		String path;
+		// Stuff in this block only happens when this activity is newly created (i.e. not a rotation)
+		if (savedInstanceState == null)
+		{
+			path = Environment.getExternalStorageDirectory().getPath();
+		} else
```
Comment (lioncash) [inline Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/activities/AddDirectoryActivity.java]: else on new line.

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/activities/GameGridActivity.java [hunk 2]
```diff
@@ -64,11 +82,45 @@ protected void onCreate(Bundle savedInstanceState)
 		}
 	}
 
+	/**
+	 * Callback from AddDirectoryActivity. Applies any changes necessary to the GameGridActivity.
+	 *
+	 * @param requestCode
+	 * @param resultCode
+	 * @param result
```
Comment (lioncash) [inline Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/activities/GameGridActivity.java]: missing descriptions

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/adapters/FileAdapter.java [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+
+	private String mPath;
+
+	private FileClickListener mListener;
+
+	/**
+	 * Initializes the dataset to be displayed, and associates the Adapter with the
+	 * Activity as an event listener.
+	 *
+	 * @param gameList
```
Comment (lioncash) [inline Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/adapters/FileAdapter.java]: This parameter doesn't exist.

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/adapters/FileAdapter.java [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+	public int getItemCount()
+	{
+		return mFileList.size();
+	}
+
+	/**
+	 * When a file is clicked, determine if it is a directory; if it is, show that new directory's
+	 * contents. If it is not, end the activity successfully.
+	 *
+	 * @param view
```
Comment (lioncash) [inline Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/adapters/FileAdapter.java]: Description.

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/adapters/FileAdapter.java [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+			mListener.finishSuccessfully();
+		}
+	}
+
+	/**
+	 * For a given directory, return a list of Files it contains.
+	 *
+	 * @param directory
+	 * @return
+	 */
```
Comment (lioncash) [inline Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/adapters/FileAdapter.java]: Descriptions.

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/activities/AddDirectoryActivity.java (hunk 1 above)
Comment (sigmabeta) [inline Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/activities/AddDirectoryActivity.java]: Good catch, I'll add this to the IntelliJ code style file.

**Comment by Sonicadvance1** [comment]
LGTM

**Pull request merged**